llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -104,14 +104,16 @@ BitVector RISCVRegisterInfo::getReservedRegs(const MachineFunction &MF) const {
   BitVector Reserved(getNumRegs());
   auto &Subtarget = MF.getSubtarget<RISCVSubtarget>();
 
-  // Mark any registers requested to be reserved as such
   for (size_t Reg = 0; Reg < getNumRegs(); Reg++) {
```

Comment:
I find it odd that this loop was looping over more than just the GPRs originally. We could maybe save some compile time by not checking every register, but this patch now requires us to check every register.

Reply:
Good point!
So I think we should iterate through all GPRs explicitly here.

Reply:
I went with iterating from `RISCV::NoRegister` to `RISCV::NUM_TARGET_REGS`, with a GPR check before calling `Subtarget.isRegisterReservedByUser`.

Reply:
To be honest, I like your first patch better. As long as RISCVSubtarget uses `std::bitset<RISCV::NUM_TARGET_REGS>` to track reserved registers, we should continue iterating all of the registers.

Reply:
Yeah I kind of agree. I reverted it back to the initial patch. I don't think calling `isRegisterReservedByUser` with non-GPRs is an issue.

Reply:
Maybe `isRegisterReservedByUser` should be replaced by `isGPRRegisterReservedByUser`. My main point was that as the number of registers increases with new extensions, this loop gets more and more iterations, making the function slower even if the new registers don't need to be reserved.